modules/python/python-grammar.ym

```diff
@@ -330,6 +334,55 @@ python_parser_loaders
           }
 
 
+optional_arrow
+          : LL_ARROW
+          |
+          ;
+
+python_option
+          : string optional_arrow string
+            {
+              PythonOption *option = python_option_string_new($1, $3);
+              python_options_add_option(last_python_options, option);
```

Comment:
this rule could return the PythonOption instance in $$ so the reference of this rule could perform python_options_add_option(), avoiding some of the duplication here.

Reply:
Also, python_options_add_option() will clone its argument, so "option" itself is leaked here.

I think we could get away with consuming the reference passed as argument, so no need for the clone. Maybe instead of clone we would only need refcounting?

Reply:
I think I will just get rid of the cloning as a whole. It is only used in the grammar, and it was only done to keep the ownerships "cleaner", but I don't think it served its purpose well.

Reply:
Done.